src/NuGet.Core/NuGet.Configuration/Settings/SettingsFile.cs

```diff
@@ -0,0 +1,240 @@
+
+            DirectoryPath = directoryPath;
+            FileName = fileName;
+            IsMachineWide = isMachineWide;
+
+            XDocument config = null;
+            ExecuteSynchronized(() =>
+            {
+                config = XmlUtility.GetOrCreateDocument(CreateDefaultConfig(), ConfigFilePath);
+            });
+
+            _xDocument = config;
+
+            _rootElement = new NuGetConfiguration(_xDocument.Root, origin: this);
+        }
+
+        /// <summary>
+        /// Gets the section with a given name.
+        /// </summary>
+        /// <param name="sectionName">name to match sections</param>
+        /// <returns>null if no section with the given name was found</returns>
+        public SettingSection GetSection(string sectionName)
+        {
+            return _rootElement.GetSection(sectionName);
+        }
+
+        /// <summary>
+        /// Adds or updates the given <paramref name="item"/> to the settings.
+        /// </summary>
+        /// <param name="sectionName">section where the <paramref name="item"/> has to be added. If this section does not exist, one will be created.</param>
+        /// <param name="item">item to be added to the settings.</param>
+        /// <returns>true if the item was successfully updated or added in the settings</returns>
+        public void AddOrUpdate(string sectionName, SettingItem item)
+        {
+            _rootElement.AddOrUpdate(sectionName, item);
+        }
+
+        /// <summary>
+        /// Removes the given <paramref name="item"/> from the settings.
+        /// If the <paramref name="item"/> is the last item in the section, the section will also be removed.
+        /// </summary>
+        /// <param name="sectionName">Section where the <paramref name="item"/> is stored. If this section does not exist, the method will throw</param>
+        /// <param name="item">item to be removed from the settings</param>
+        /// <remarks> If the SettingsFile is a machine wide config this method will throw</remarks>
+        public void Remove(string sectionName, SettingItem item)
+        {
+            _rootElement.Remove(sectionName, item);
+        }
+
+        /// <summary>
+        /// Flushes any in-memory updates in the SettingsFile to disk.
+        /// </summary>
+        public void SaveToDisk()
+        {
+            if (IsDirty)
+            {
+                ExecuteSynchronized(() =>
+                {
+                    FileSystemUtility.AddFile(ConfigFilePath, _xDocument.Save);
+                });
+
+                IsDirty = false;
+                SettingsChanged?.Invoke(this, EventArgs.Empty);
+            }
+        }
+
+        internal bool IsEmpty() => _rootElement == null || _rootElement.IsEmpty();
+
+        internal bool TryGetSection(string sectionName, out SettingSection section)
+        {
+            var result = _rootElement.Sections.TryGetValue(sectionName, out var parsedSection);
+            section = parsedSection;
```

Comment:
you can directly use `parsedSection` while reading from rootElement.

Reply:
the compiler complained about this because of the types, that's why I had to have an extra variable.